File: frontend/src/types/product.ts
export interface Product {
  id: number
  productCode: string
  name: string
  price: number
  stockQuantity: number
  createdAt: string
}

export interface ProductRequest {
  productCode: string
  name: string
  price: number
  stockQuantity: number
}

export interface ProductPageResponse {
  items: Product[]
  totalCount: number
  page: number
  pageSize: number
  totalPages: number
}

File: backend/Models/Product.cs
namespace StoreWeb.Api.Models;

public class Product
{
    public int Id { get; set; }
    public string ProductCode { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public decimal Price { get; set; }
    public int StockQuantity { get; set; }
    public string? ImageUrl { get; set; }
    public DateTime CreatedAt { get; set; }
}
